Sources/Datadog/URLSessionAutoInstrumentation/Interception/TracingHeaderTypesProvider/TracingHeaderTypesProvider.swift

```diff
@@ -0,0 +1,33 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal struct TracingHeaderTypesProvider {
+    private let firstPartyHosts: FirstPartyHosts
+
+    init(
+        firstPartyHosts: FirstPartyHosts
+    ) {
+        self.firstPartyHosts = firstPartyHosts
+    }
+
+    func tracingHeaderTypes(for url: URL?) -> Set<TracingHeaderType> {
+        return firstPartyHosts.compactMap { item -> Set<TracingHeaderType>? in
+            let regex = "^(.*\\.)*\(NSRegularExpression.escapedPattern(for: item.key))$"
+            if url?.host?.range(of: regex, options: .regularExpression) != nil {
+                return item.value
+            }
+            if url?.absoluteString.range(of: regex, options: .regularExpression) != nil {
+                return item.value
+            }
```

Comment:
I wonder why do we need to check the `absoluteString`? Unlike `host`, the absolute string may include many arbitrary elements (e.g. query path), which makes it way harder to construct a valid regex.

In practise, I managed to build this false positive example:
```swift
func testFalsePositive() {
    let provider = TracingHeaderTypesProvider(
        firstPartyHosts: [
            "example.com": [.dd, .b3m],
        ]
    )

    let url = URL(string: "http://foo.com/something.example.com")

    XCTAssertEqual(provider.tracingHeaderTypes(for: url), [])
}

// XCTAssertEqual failed: ("[Datadog.TracingHeaderType.b3m, Datadog.TracingHeaderType.dd]") is not equal to ("[]")
```

Reply:
Yeah - this is not a valid domain. I will think about moving sanitizer inside the new `FirstPartyHosts`